angular-chart.js

```diff
@@ -321,12 +325,16 @@
       });
     }
 
-    function setLegend (elem, chart) {
+    function setLegend (elem, chart, scope) {
       var $parent = elem.parent(),
-          $oldLegend = $parent.find('chart-legend'),
-          legend = '<chart-legend>' + chart.generateLegend() + '</chart-legend>';
-      if ($oldLegend.length) $oldLegend.replaceWith(legend);
-      else $parent.append(legend);
+          $oldLegend = $parent.find('chart-legend');
+      if (scope.legend) {
+        var legend = '<chart-legend>' + chart.generateLegend() + '</chart-legend>';
+        if ($oldLegend.length) $oldLegend.replaceWith(legend);
+        else $parent.append(legend);
+      } else {
+        $oldLegend.remove();
```

Comment:
And what happened if there wasn't an `$oldLegend`? Should probably check that it exists before calling a function on it, seems brittle.